src/app/logproc/hashfarm.ml

```diff
@@ -0,0 +1,51 @@
+open Core
+
+let all_pairs (xs : 'a list) (ys : 'b list) : ('a * 'b) list =
+  List.bind ~f:(fun x -> List.map ~f:(fun y -> (x, y)) ys) xs
+
+let all_names =
+  lazy
+    ( all_pairs Hashfarm_wordlists.adjectives
+        (all_pairs Hashfarm_wordlists.colors Hashfarm_wordlists.animals)
+    |> Array.of_list
+    |> Array.map ~f:(fun (a, (b, c)) -> a ^ " " ^ b ^ " " ^ c) )
+
+let used_names : (string, string) Hashtbl.t = Hashtbl.create (module String)
+
+let select_name x =
```

Comment:
Can this be named hash?

Reply:
it isn't a hash

Reply:
derp, I misread your comment. yes, it can be.